core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -982,6 +982,12 @@ class GroupCoordinator(val brokerId: Int,
   ): Option[Errors] = {
     if (group.is(Dead)) {
       Some(Errors.COORDINATOR_NOT_AVAILABLE)
+    } else if (!isTransactional && generationId < 0 && group.is(Empty)) {
```

Comment:
I am not sure I follow the reason for the transactional check. Can you clarify?

Reply:
I thought that transactional commits should never comes with a generation < 0. I suppose that this is wrong. I suppose that a producer could commit transactional offsets without using a consumer group. Is this right?

Reply:
It seems to be allowed at the moment, though that is definitely not a common case.

Reply:
Removed it.